File: Python3/Medium/WrongAnswer/number-of-pairs-of-strings-with-concatenation-equal-to-target_4.py
class Solution:
    def numOfPairs(self, nums: List[str], target: str) -> int:
        
        d = collections.defaultdict(int)
        res = 0
        for num in nums:
            d[num] += 1
        for num in nums:
            if target.startswith(num):
                res += d[target[len(num):]]
            if target.endswith(num):
                res += d[target[:-len(num)]]
            if target.startswith(num) and target.endswith(num):
                res -= 1
        return res // 2

File: Java/Medium/Accepted/number-of-pairs-of-strings-with-concatenation-equal-to-target_4.java
class Solution {
    public int numOfPairs(String[] nums, String target) {
        
        int res = 0;
        for (int i = 0; i < nums.length; i++) {
            for (int j = 0; j < nums.length; j++) {
                if (i != j && nums[i].concat(nums[j]).equals(target)) {
                    res++;
                }
            }
        }
        return res;
    }
}